Classes/DataProcessing/GalleryProcessor.php

```diff
@@ -184,19 +184,30 @@ protected function prepareGalleryData()
                 $fileObj = $this->fileObjects[$fileKey] ?? null;
 
                 if ($fileObj) {
+                    $fileExtension = $this->processorConfiguration['fileExtension'] ?? null;
+
                     if ($fileObj['properties']['type'] === 'image') {
                         $image = $this->getImageService()->getImage((string)$fileObj['properties']['fileReferenceUid'], null, true);
-                        $fileObj = $this->getFileUtility()->processFile($image, $this->mediaDimensions[$fileKey] ?? []);
+                        $fileObj = $this->getFileUtility()->processFile(
+                            $image,
+                            array_merge(
+                                ['fileExtension' => $fileExtension],
+                                $this->mediaDimensions[$fileKey] ?? []
+                            )
+                        );
 
                         if (isset($this->processorConfiguration['autogenerate.']['retina2x'],
                             $fileObj['properties']['dimensions']['width']) &&
                             (int)$this->processorConfiguration['autogenerate.']['retina2x'] === 1) {
                             $fileObj['urlRetina'] = $this->getFileUtility()->processFile(
                                 $image,
-                                [
-                                    'width' => $fileObj['properties']['dimensions']['width'] * FileUtility::RETINA_RATIO,
-                                    'height' => $fileObj['properties']['dimensions']['height'] * FileUtility::RETINA_RATIO,
-                                ]
+                                array_merge(
+                                    ['fileExtension' => $fileExtension],
```

Comment:
array_merge is overkill here, we could simply add to main array `fileExtension`

Reply:
@twoldanski, you're right ... buuuuut: what about other arguments that might be passed from TS? Can you think of anything that might come in handy (later)?

Reply:
@kitzberger For now we do not have any more params passed, we do not need to complicate for "the future". 
I think File/Gallery processing need some love, and needs more refactoring later on.